connector/openshift/openshift_test.go

```diff
@@ -184,6 +184,57 @@ func TestCallbackIdentity(t *testing.T) {
 	expectEquals(t, identity.Groups[0], "users")
 }
 
+func TestCheckOAuth2EndpointsAvailability(t *testing.T) {
+	s := newTestServer(map[string]interface{}{
+		"/oauth/authorize": nil,
+		"/oauth/token": nil,
+	})
+	defer s.Close()
+
+	hostURL, err := url.Parse(s.URL)
+	expectNil(t, err)
+
+	_, err = http.NewRequest("GET", hostURL.String(), nil)
+	expectNil(t, err)
+
+	h, err := newHTTPClient(true, "")
+	expectNil(t, err)
```

Comment:
I would still maintain that this doesn't make sense. All it does is to ensure that the failure in the `...Failure` method below is not spurious/caused by a test server misconfiguration. However, you already exercise the positive path in this method, so this isn't much of a danger. To be on the safe side, you could exercise both the positive and the negative path in the same test method, by first executing these steps, then calling `s.Close()`, and then running l.234-235 below. This would eliminate the failure scenario where the test server setup diverges between the two methods, being correctly configured in the positive test case and erroneously configured in the negative test case. But that is really paranoia more than anything else.

Reply:
I have removed it now, since I also have not added it to the Refresh related tests and merged the two cases (success and failure) together in a single test